### PR TITLE
Refactor a few selectors.

### PR DESCRIPTION
**Pull request by XhmikosR**
1. move a couple of them before the more specific ones.
2. change nesting to be under the same parent selector
3. use the ampersand more

### CODE REVIEW

File: assets/scss/_component-examples.scss [hunk 1]
```diff
@@ -170,31 +170,36 @@
...
+    }
+  }
+
+  h1,
+  h2,
+  h3,
+  h4,
+  h5,
+  h6 {
+    margin: 0;
```
Comment (mdo) [inline assets/scss/_component-examples.scss]: I _think_ we can change this to `margin-top: 0;` and `margin-bottom: 0;`. Can't imagine we're adding horizontal margin to any headings.

File: scss/_modal.scss [hunk 2]
```diff
@@ -26,16 +26,10 @@
   // gnarly iOS Safari bug: https://bugs.webkit.org/show_bug.cgi?id=158342
   // See also https://github.com/twbs/bootstrap/issues/17695
 
-  // When fading in the modal, animate it to slide down
-  &.fade .modal-dialog {
-    @include transition($modal-transition);
-    transform: translate(0, -25%);
+  .modal-open & {
```
Comment (mdo) [inline scss/_modal.scss]: In general I'm not a fan of this type of nesting—usually ends up more difficult to parse in more complicated rulesets.

File: scss/_code.scss [hunk 3]
```diff
@@ -31,7 +31,7 @@ kbd {
   @include border-radius($border-radius-sm);
   @include box-shadow($kbd-box-shadow);
 
-  kbd {
+  & & {
```
Comment (mdo) [inline scss/_code.scss]: This is weird at first, but probably fine.

File: scss/_modal.scss (hunk 2 above)
Comment (XhmikosR) [reply scss/_modal.scss]: I think we need to keep this; it doesn't complicate things a lot and along with the complexity rule we should be fine. I plan to enable that later too.

I removed the `& &` instances.

**Review by mdo — APPROVED** [review]

**Review by andresgalante — APPROVED** [review]
@XhmikosR sorry that I couldn't help much with this one. Thanks for doing it. You made a linter happy :)

**Comment by XhmikosR** [comment]
@andresgalante: this isn't quite ready since I haven't enabled the rule yet. Still waiting for your patch so that we enable the rule. :)

**Comment by andresgalante** [comment]
:p lol I thought it was! ok I'll continue with it

**Comment by XhmikosR** [comment]
@andresgalante: I'm gonna merge this as is, and we should tackle the 2 remaining stylelint rules later.

**Pull request merged**